### ISSUE TITLE
socket ssl wrapping error: hostname doesn't match

### ISSUE THREAD

**Issue by eddie-dunn** [cannatag/ldap3#180]
When I try to create a Server() with tls enabled I get the following error:

```
ldap3.core.exceptions.LDAPSocketOpenError: socket ssl wrapping error: hostname doesn't match
```

That's not very useful; which hostname doesn't match what?

Following the otherwise excellent [tutorial](http://ldap3.readthedocs.io/tutorial.html#establishing-a-secure-connection), here's the code that produced the error:

```
import ssl
from ldap3 import Server, ALL, Tls, Connection
LDAP_ADDR = 'internal.address.com'
tls_config = Tls(validate=ssl.CERT_REQUIRED, version=ssl.PROTOCOL_TLSv1)
server = Server(LDAP_ADDR, use_ssl=True, tls=tls_config)
conn = Connection(server)
conn.open()
```

**Comment by cannatag** [comment]
Hi which Python version ate you using? That message could be emitted by the python ssl standard library.

Is your server configured for ssl with a valid certificate? If set validate=ssl.CERT_OPTIONAL does it works?

Giovanni

**Comment by eddie-dunn** [comment]
Sorry, forgot to say that I am using Python3.

ssl.CERT_OPTIONAL does not work either. I think I know why the ssl certification fails; that's more or less expected, since my company can't be arsed to get real certificates.

The reason I opened the issue is that I think the exception lacks information.

**Comment by cannatag** [comment]
Hi Eddie, you're right. I'll try to give a better explanation of the error in the exception. Hope to have a new version of the code for the next release of ldap3.

Bye,
Giovanni

**Comment by cannatag** [comment]
Hi,
now the exception prints out the whole certificate and the list of names that are checked::
```
Traceback (most recent call last):
  File "C:\Python\Workspace\test5\test1.py", line 9, in <module>
    conn.open()
  File "c:\Python\OpenSource\ldap3\ldap3\strategy\sync.py", line 55, in open
    BaseStrategy.open(self, reset_usage, read_server_info)
  File "c:\Python\OpenSource\ldap3\ldap3\strategy\base.py", line 134, in open
    raise exception_history[0][1](exception_history[0][2])
ldap3.core.exceptions.LDAPSocketOpenError: socket ssl wrapping error: certificate {'subjectAltName': (('IP Address', '192.168.137.101'),), 'subject': ((('organizationName', 'EDIR-LAB'),), (('commonName', 'edir1.hyperv'),)), 'issuer': ((('organizationalUnitName', 'Organizational CA'),), (('organizationName', 'EDIR-LAB'),)), 'notAfter': 'Mar  2 18:45:44 2018 GMT', 'crlDistributionPoints': ('http://192.168.137.101:8028/crl/one.crl', 'ldap://192.168.137.101:389/CN=One,CN=One%20-%20Configuration,CN=CRL%20Container,CN=Security', 'https://192.168.137.101:8030/crl/one.crl', 'ldaps://192.168.137.101:636/CN=One,CN=One%20-%20Configuration,CN=CRL%20Container,CN=Security'), 'notBefore': 'Mar  2 18:45:44 2016 GMT', 'version': 3, 'serialNumber': '021C11FFA8456321C7477DC891CA14E947C0B00ED82C9751B18E09A04BF902022517CF\\\n'} doesn't match any name in ['edir1.hyperv'] 
```

With this information is much easier to fix name matching problems. Do you think it's ok?

You can check the latest code in dev. It will be included in the next release of ldap3.

thanks,
Giovanni

**Comment by cannatag** [comment]
released in 1.3.3

**Comment by eddie-dunn** [comment]
Looks good to me. Thanks.
